### PR TITLE
[wasm-ctor-eval] Add support for multivalue serialization and a quiet mode

### PR DESCRIPTION
**Pull request by kripken**
Simply loop over the values and use `tuple.make`.

This also adds a lit test for ctor-eval. I found that the problem blocking us
before was the logging, which confuses the update script. As this test at
least does not require that logging, this PR adds a `--quiet` flag that
disables the logging, and then a lit test just works.

Separate PRs for the two changes are possible, but only together can they
both be tested properly.

Diff without whitespace is smaller.

### CODE REVIEW

File: src/tools/wasm-ctor-eval.cpp [hunk 1]
```diff
@@ -613,11 +613,23 @@ struct CtorEvalExternalInterface : EvallingModuleRunner::ExternalInterface {
...
+      }
+      return Builder(*wasm).makeTupleMake(children);
+    }
     assert(values.size() == 1);
     return getSerialization(values[0], possibleDefiningGlobal);
   }
 };
 
+// Whether to emit informative logging to stdout about the eval process.
+static bool quiet = false;
```
Comment (tlively) [inline src/tools/wasm-ctor-eval.cpp:631]: Another solution might be to add a log prefix and give the lit auto update script an option to specify that prefix so the script can recognize and emit checks for log lines.

**Review by tlively — APPROVED** [review]

File: src/tools/wasm-ctor-eval.cpp [hunk 2]
```diff
@@ -613,11 +613,23 @@ struct CtorEvalExternalInterface : EvallingModuleRunner::ExternalInterface {
 
   Expression* getSerialization(const Literals& values,
                                Name possibleDefiningGlobal = Name()) {
+    if (values.size() > 1) {
+      // We do not support multivalues in defining globals, which store GC refs.
+      assert(possibleDefiningGlobal.isNull());
```
[the remark this answers is not in the document]
Comment (kripken) [reply src/tools/wasm-ctor-eval.cpp:618]: It's a global in which the definition of a GC allocation lives, when we serialize one. See

https://github.com/WebAssembly/binaryen/blob/653923644c7517dae91553e6185962b5dcb9eb3e/src/tools/wasm-ctor-eval.cpp#L478-L485

File: src/tools/wasm-ctor-eval.cpp (hunk 1 above)
Comment (kripken) [reply src/tools/wasm-ctor-eval.cpp:631]: I see, thanks. That might be useful when we translate the existing tests to lit, which do care about the logging.

File: src/tools/wasm-ctor-eval.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (tlively) [reply src/tools/wasm-ctor-eval.cpp:618]: Thanks!

**Pull request merged**